crates/flox/src/commands/channel.rs

```diff
@@ -76,122 +88,131 @@ pub struct Search {
 impl Search {
     pub async fn handle(self, flox: Flox) -> Result<()> {
         subcommand_metric!("search");
-        let channels = flox
-            .channels
-            .iter()
-            .filter_map(|entry| {
-                if HIDDEN_CHANNELS.contains_key(&*entry.from.id) {
-                    None
-                } else if DEFAULT_CHANNELS.contains_key(&*entry.from.id) {
-                    Some((ChannelType::Flox, entry))
-                } else {
-                    Some((ChannelType::User, entry))
-                }
-            })
-            .sorted_by(|a, b| Ord::cmp(a, b));
 
-        // Create `registry` parameter for `pkgdb`
-        let mut priority: Vec<String> = Vec::new();
-        let mut inputs = serde_json::Map::new();
-        for (_, entry) in channels {
-            // TODO: handle `subtrees` and `stabilities`
-            priority.push(entry.from.id.to_string());
-            inputs.insert(
-                entry.from.id.to_string(),
-                json!({"from": entry.from.to_string()}),
-            );
+        if self.search_term.is_none() {
+            bail!(
+                "'flox search' requires at least one search term, run 'flox search -h' for more information."
+            )
         }
-        let registry = json!({
-            "priority": priority,
-            "inputs": inputs,
-            "default": json!({
-                "subtrees": null, // TODO
-                "stabilities": null,  // TODO
-            })
-        });
-
-        // Create `query` parameter for `pkgdb`
-        let query = match self.search_term {
-            Some(search_term) => {
-                let mut query = serde_json::Map::new();
-                query.insert("match".to_string(), json!(search_term));
-                query
-            },
-            None => serde_json::Map::new(),
-        };
 
-        let params = json!({
-            "registry": registry,
-            "query": query,
-            // These defaults are set by `pkgdb` but are shown here
-            // for visibility.
-            // "semver": { "preferPreReleases": false }
-            // "systems": ["x86_64-linux", ...]
-        });
-
-        // TODO: I have no idea how to check verbosity here.
-        //// By default we run with `--quiet`, but if verbose is set then
-        //// stream stderr.
-        //let mut maybe_quiet: Vec<String> = Vec::new();
-        //if !self.verbose {
-        //    maybe_quiet.push("--quiet".to_string());
-        //}
-        //maybe_quiet.push("--quiet".to_string());
-        let maybe_quiet: Vec<String> = vec!["--quiet".to_string()];
+        let search_params = construct_search_params(&self.search_term, &flox)?;
+        let search_params_json = serde_json::to_string(&search_params)?;
 
         let output = Command::new(PKGDB_BIN.as_str())
```

Comment:
Seems like `PKGDB_BIN` and a function like `call_pkgdb` (or even just name it `search`, the CLI doesn't need to know that we're using pkgdb) should live in `flox-rust-sdk`, and then from the CLI crate perspective, you just pass in search params and get back results.

Reply:
I'm not opposed to that, but I'll leave it to a later PR when we overhaul the error handling since that's likely to determine a large part of the interface.

Reply:
I'm also working on `flox show` at the same time and there might be opportunities for re-use, so I'd like to wait to settle on a search interface until I have a better handle on how `flox show` will shake out.